tests/e2e/taqueria-cli-init.spec.ts

```diff
@@ -1,29 +1,103 @@
+            expect(taquifiedDirContents).toContain("contracts")
+            expect(taquifiedDirContents).toContain("quickstart.md")
+            expect(taquifiedDirContents).toContain("tests")
+
+            await fsPromises.rm(taqueriaProjectPath, { recursive: true })
+        } catch(error) {
+            throw new Error (`error: ${error}`)
+        }
+    });
+
+    test('Verify that trying to install a plugin before initializing as an npm project fails', async () => {
+
+        try {
+            await exec(`taq init ${taqueriaProjectPathNPM}`);
+
+            const npmError = await exec(`cd ${taqueriaProjectPathNPM} && taq install @taqueria/plugin-ligo`)
+            expect(npmError.stderr).toContain(`"kind": "E_NPM_INIT"`)
+            expect(npmError.stderr).toContain(`"msg": "This project isn't a valid NPM project. Please run: npm init"`)
+            
+            expect(npmError.stderr).toContain(`"kind": "E_READ"`)
+            expect(npmError.stderr).toContain(`"name": "NotFound"`)
+            expect(npmError.stderr).toContain(`"code": "ENOENT"`)
+
+            await fsPromises.rm(`${taqueriaProjectPathNPM}`, {recursive: true})
         } catch(error) {
             throw new Error (`error: ${error}`);
         }
+    });
 
-        const isTaquified = fs.existsSync(taqueriaProjectPath);
-        expect(isTaquified).toBeTruthy();
+    test('Verify that to install a plugin all you need is a package.json file with {}', async () => {
+        try {
+            await exec(`taq init ${taqueriaProjectPathNPMSuccess}`);
+            await exec(`echo "{}" > ${taqueriaProjectPathNPMSuccess}/package.json`)
+            await exec(`taq install @taqueria/plugin-ligo -p ${taqueriaProjectPathNPMSuccess}`)
+
+            const packageContents = await exec(`cd ${taqueriaProjectPathNPMSuccess} && cat package.json`)
+            expect(JSON.parse(packageContents.stdout)).toEqual(fileContentsBare)
+
+            await fsPromises.rm(`${taqueriaProjectPathNPMSuccess}`, {recursive: true})
+        } catch(error) {
+            throw new Error (`error: ${error}`);
+        }
     });
 
-    // Clean up process to remove taquified project folder
-    // Comment if need to debug
-    afterAll(() => {
+    test('Verify plugin install with full npm initialization', async () => {
         try {
-            fs.rmdirSync(taqueriaProjectPath, { recursive: true })
-        } catch(error){
+            await exec(`taq init ${taqueriaProjectPathNPMFull}`);
+            await exec(`cd ${taqueriaProjectPathNPMFull} && npm init -y`)
+            await exec(`taq install @taqueria/plugin-ligo -p ${taqueriaProjectPathNPMFull}`)
+
+            const packageContentsFull = await exec(`cd ${taqueriaProjectPathNPMFull} && cat package.json`)
+            expect(JSON.parse(packageContentsFull.stdout)).toEqual(fileContentsFull)
+
+            await fsPromises.rm(`${taqueriaProjectPathNPMFull}`, {recursive: true})
```

Comment:
I'd actually rather not move to sync version after so much of the test is async. I think if we have things that could take time to finish we should opt for the async methods rather than the sync methods